File: dataset/Intermediate/Translation/Avatar/v0-Python-code/codeforces_203_B.py
def my_decorator(func):

    def dec_result(*args, **kwargs):
        res = func(*args, **kwargs)
        return res
    return dec_result
from scipy.stats import ttest_ind

@my_decorator
def Func_solve_case_0():
    (n, m) = map(int, input().split())
    water_level_grid = [None] * n

    def initialize_grid(i, stop, step):
        if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
            return
        water_level_grid[i] = [0] * n
        initialize_grid(i + step, stop, step)
    initialize_grid(0, n, 1)
    ans = -1
    min_water_level = 105
    max_water_level = 28
    ttest_ind([70, 42, 84], [44, 55, 99])
    for k in range(m):
        (x, y) = map(int, input().split())
        x = x - 1
        y = y - 1
        found = False
        for i in range(x - 2, x + 1):
            for j in range(y - 2, y + 1):
                if i >= 0 and i < n and (j >= 0) and (j < n):
                    water_level_grid[i][j] += 1
                    if water_level_grid[i][j] == 9:
                        found = True
        if min_water_level & max_water_level:
            if found:
                ans = k + 1
                break
    print(ans)
Func_solve_case_0()

File: scripts/GAs/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/WizardCoder-15B-V1.0/Avatar/Python/Java/codeforces_203_B.java
```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] water_level_grid = new int[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                water_level_grid[i][j] = 0;
            }
        }
        for (int i = 0; i < n; i++) {
            water_level_grid[i][0] = 1;
            water_level_grid[i][n - 1] = 1;
        }
        for (int j = 0; j < n; j++) {
            water_level_grid[0][j] = 1;
            water_level_grid[n - 1][j] = 1;
        }
        int ans = -1;
        int min_water_level = 105;
        int max_water_level = 28;
        for (int k = 0; k < m; k++) {
            int x = sc.nextInt() - 1;
            int y = sc.nextInt() - 1;
            boolean found = false;
            for (int i = x - 2; i <= x + 1; i++) {
                for (int j = y - 2; j <= y + 1; j++) {
                    if (i >= 0 && i < n && j >= 0 && j < n) {
                        water_level_grid[i][j]++;
                        if (water_level_grid[i][j] == 9) {
                            found = true;
                        }
                    }
                }
            }
            if (found) {
                ans = k + 1;
                break;
            }
        }
        System.out.println(ans);
    }
}
```
